thoth-export-server/src/marc21/marc21record_thoth.rs

```diff
@@ -182,22 +170,39 @@ impl Marc21Entry<Marc21RecordThoth> for Work {
         }
 
         // 100 and 700 - contributors
-        for contributor_field in contributor_fields(&self.contributions)? {
-            builder.add_field(contributor_field)?;
+        let contributor_fields = contributor_fields(&self.contributions)?;
+        for contributor_field in &contributor_fields {
+            builder.add_field(contributor_field.clone())?;
         }
 
         // 245 – title
-        FieldRepr::from((b"245", "00"))
-            .add_subfield(b"a", self.title.clone().into_bytes())
-            .and_then(|f| f.add_subfield(b"h", b"[electronic resource] :"))
-            .and_then(|f| {
-                if let Some(subtitle) = self.subtitle.clone() {
-                    f.add_subfield(b"b", subtitle.into_bytes())
-                } else {
-                    Ok(f)
-                }
-            })
-            .and_then(|f| f.add_subfield(b"c", contributors_string(&self.contributions).as_bytes()))
+        let added_entry = match contributor_fields.iter().any(|c| c.get_tag() == b"100") {
+            true => "1",
+            false => "0",
+        };
+        let mut nonfiling_char_count = "0";
```

Comment:
This could probably be abstracted into a function, especially so we can unit test it.

It'd also be good to have all language prefixes as a list so we can update them easily, you can then iterate the list and just return `prefix.len()`.